### PR TITLE
Bump dependencies to latest. Fix `DOJO_CLI` typing.

### PR DESCRIPTION
**Pull request by tomdye**
**Type:** feature

The following has been addressed in the PR:

* [x] There is a related issue
* [ ] All code matches the [style guide](https://github.com/dojo/meta/blob/master/STYLE.md)
* [ ] Unit or Functional tests are included in the PR

**Description:**

Bumps dependencies to latest as they were getting outdated.
Switches to the typings shipped with chalk.

Requires: https://github.com/dojo/interfaces/pull/88
Resolves: https://github.com/dojo/cli/issues/188
Resolves: https://github.com/dojo/cli/issues/191

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -31,49 +31,45 @@
     "typings.json"
   ],
   "dependencies": {
-    "@dojo/cli-create-app": "~0.3.0",
```
Comment (agubler) [inline package.json]: Should we be removing this as part of this PR?

File: package.json [hunk 2]
```diff
@@ -31,49 +31,45 @@
...
     "@dojo/core": "~0.3.0",
     "@dojo/has": "~0.1.1",
     "@dojo/shim": "~0.2.3",
-    "chalk": "^1.1.3",
-    "cliui": "^3.2.0",
-    "cross-spawn": "^4.0.0",
-    "david": "^9.0.0",
+    "chalk": "^2.3.0",
```
Comment (agubler) [inline package.json]: do you think we should pin these dependencies?

File: package.json (hunk 1 above)
Comment (tomdye) [reply package.json]: my bad

**Comment by tomdye** [comment]
Closing in favour of https://github.com/dojo/cli/pull/200 which adds command list functionality also

**Pull request closed without merging**